### PR TITLE
RUN: Support code coverage via grcov

### PR DESCRIPTION
**Pull request by mchernyavsky**
grcov [repo](https://github.com/mozilla/grcov)

Closes https://github.com/intellij-rust/intellij-rust/issues/1230.

image

TODO: add a GIF

### CODE REVIEW

**Comment by Undin** [comment]
@mchernyavsky Rebase onto master when #3632 will be merged and make gradle clean
It should fix failed tests

File: coverage/src/main/kotlin/org/rust/coverage/GrcovRunner.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    companion object {
+        private val LOG: Logger = Logger.getInstance(GrcovRunner::class.java)
+
+        private fun updateCoverageView(env: ExecutionEnvironment, coverageReport: File) {
+            val runConfiguration = env.runProfile as RunConfigurationBase<*>
+            CoverageHelper.resetCoverageSuit(runConfiguration)
+            val coverageEnabledConfiguration = CoverageEnabledConfiguration.getOrCreate(runConfiguration)
+            val coverageFilePath = coverageEnabledConfiguration.coverageFilePath ?: return
+            try {
+                FileUtil.copy(coverageReport, File(coverageFilePath))
```
Comment (amakeev) [inline coverage/src/main/kotlin/org/rust/coverage/GrcovRunner.kt]: is it possible to move copying outside of UI thread?

**Comment by Undin** [comment]
@mchernyavsky Could you add note about new module [here](https://github.com/intellij-rust/intellij-rust/blob/master/CONTRIBUTING.md#modules)?

File: coverage/src/main/kotlin/org/rust/coverage/GrcovRunner.kt [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import org.rust.stdext.toPath
+import java.io.File
+
+class GrcovRunner : DefaultProgramRunner() {
+    override fun getRunnerId(): String = "GrcovRunner"
+
+    override fun canRun(executorId: String, profile: RunProfile): Boolean =
+        executorId == CoverageExecutor.EXECUTOR_ID &&
+            profile is CargoCommandConfiguration &&
+            profile.clean().ok != null
```
[the remark this answers is not in the document]
Comment (ortem) [reply coverage/src/main/kotlin/org/rust/coverage/GrcovRunner.kt]: @Undin what do you think? Should we show the coverage button on `build` (and others) run configurations?

**Comment by Undin** [comment]
@mchernyavsky Rebase onto master, please
Take into account changes in gradle script related to changes in 192 platform. See 5935b2eba419bc8aa75081bb6ba87a7c8a3ba468

**Comment by mchernyavsky** [comment]
@ortem I have rebased it.

**Comment by mchernyavsky** [comment]
bors retry

**Pull request merged**